bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -338,6 +408,82 @@ async def nominate_command(self, ctx: Context, user: MemberOrUser, *, reason: st
 
         await self._nominate_user(ctx, user, reason)
 
+    @app_commands.checks.has_any_role(*STAFF_ROLES)
+    async def _nominate_context_callback(self, interaction: discord.Interaction, message: discord.Message) -> None:
+        """Create or update a nomination for the author of the selected message."""
+        if message.author.bot:
+            await interaction.response.send_message(
+                ":x: I'm afraid I can't do that. Only humans can be nominated.",
+                ephemeral=True
+            )
+            return
+
+        if isinstance(message.author, Member) and any(role.id in STAFF_ROLES for role in message.author.roles):
+            await interaction.response.send_message(
+                ":x: Nominating staff members, eh? Here's a cookie :cookie:",
+                ephemeral=True
+            )
+            return
+
+        maybe_nom = await self.api.get_nomination_reason(message.author.id, interaction.user.id)
+
+        if maybe_nom:
+            nomination, reason = maybe_nom
+
+            reason += f"\n\n{message.jump_url}"
+
+            await self.api.edit_nomination_entry(
+                nomination.id,
+                actor_id=interaction.user.id,
+                reason=reason
+            )
```

Comment:
Do we need to handle the reason being too long @jb3?

Reply:
I ignorantly hoped people would just not add a shit ton of messages, but yes we should handle this, PR in a second.